### PR TITLE
make json-api's scopt better-typed by passing JDBC drivers implicitly

### PR DESCRIPTION
**Pull request by S11001001**
This solves two warts in the code:

- the `validate`/`createUnsafe` double-parse, because scopt doesn't let you flatMap;
- the non-`JdbcConfig` sub-configs appeared to need to know the JDBC drivers when they really don't, because of a quirk in the inherited implementation (a "foolish consistency", literally)

That coherence of `scopt.Read` instances calls for all its dependencies to be coherent leads us to treat `supportedJdbcDriverNames` as a nullary typeclass instance.  This is a nullary typeclass by the same justification as `SupportedJdbcDriver`; see scaladoc there for more.

And we solve the latter problem by...adding a type parameter, how else.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by realvictorprm — APPROVED** [review]
Looks good to me, although it took some time for me to get through how it is working now.

**Review by cocreature — APPROVED** [review]

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala [hunk 1]
```diff
@@ -207,9 +208,8 @@ private[http] object WebsocketConfig extends ConfigCompanion[WebsocketConfig]("W
   )
 
   override def create(
-      x: Map[String, String],
-      supportedJdbcDriverNames: Set[String],
-  ): Either[String, WebsocketConfig] =
+      x: Map[String, String]
+  )(implicit readCtx: DummyImplicit): Either[String, WebsocketConfig] =
```
Comment (stefanobaghino-da) [inline ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:212]: Why is this necessary? Can we get away with renaming `create` to something else instead of introducing a dummy implicit?

**Review by stefanobaghino-da — APPROVED** [review]

File: ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala (hunk 1 above)
Comment (S11001001) [reply ledger-service/http-json-cli/src/main/scala/com/daml/http/Config.scala:212]: There has to be something there, to match the abstract signature.

My first instinct was initially to leave the argument where it was, but passing `()` explicitly was a little inelegant, but more importantly, it wouldn't mesh as well with supplying the `Read` instances.
